### PR TITLE
Drop testing obsolete SSH DSA key type

### PR DESCRIPTION
**Pull request by martinpitt**
DSA got [deprecated in OpenSSH 7](https://www.openssh.com/legacy.html) , and should not be used any more. The upcoming libssh 0.10 also drops support for it. Move to current EC keys as alternative to RSA keys.

libssh 0.10/master also [has a regression with DSA keys](https://gitlab.com/libssh/libssh-mirror/-/issues/145) which motivated this change.

I built  libssh master, and ran the unit tests against it:

    LD_LIBRARY_PATH=~/upstream/libssh/build/lib/ ./test-sshbridge

they pass with this branch.

@cryptomilk FYI

### CODE REVIEW

**Comment by jelly** [comment]
Should this have a `release-note` when it's merged?

**Comment by martinpitt** [comment]
@jelly : No, this doesn't change any runtime behaviour, just tests. Of course once Fedora and other distros update to libssh 0.10, this will drop DSA support -- but that's kinda outside of what we should announce, I think?

**Comment by jelly** [comment]
> @jelly : No, this doesn't change any runtime behaviour, just tests. Of course once Fedora and other distros update to libssh 0.10, this will drop DSA support -- but that's kinda outside of what we should announce, I think?

Aha, nevermind then.

File: src/ssh/test-sshbridge.c [hunk 1]
```diff
@@ -699,7 +699,7 @@ test_echo_large (TestCase *tc,
 static const gchar MOCK_RSA_FP[] = "SHA256:XQ8a7zGxMFstDrGecBRUP9OMnOUXd/T3vkNGtYShs2w";
 #define SSH_PUBLICKEY_HASH_NAME "SHA256"
 
-#define MOCK_DSA_PUB_KEY "ssh-dss AAAAB3NzaC1kc3MAAACBAIK3RTEWBw+rAPcYUM2Qq4kEw59gXpUQ/WvkdeY7QDO64MHaaorySj8xsraNudmQFh4xb/i5Q1EMnNchOFxtilfU5bUJgdTvetyZEWFL+2HxqBs8GaWRyB1vtSFAw3GO8VUEnjF844N3dNyLoc0NX8IvzwNIaQho6KTsueQlG1X9AAAAFQCXUl4a5UvElL4thi/8QlxR5PtEewAAAIBqNpl5MTBxKQu5jT0+WASa7pAqwT53ofv7ZTDIEokYRb57/nwzDgkcs1fsBRrI6eczJ/VlXWwKbsgkx2Nh3ZiWYwC+HY5uqRpDaj3HERC6LMn4dzdcl29fYeziEibCbRjJX5lZF2vIaA1Ewv8yT0UlunyHZRiyw4WlEglkf/NITAAAAIBxLsdBBXn+8qEYwWK9KT+arRqNXC/lrl0Fp5YyxGNGCv82JcnuOShGGTzhYf8AtTCY1u5oixiW9kea6KXGAKgTjfJShr7n47SZVfOPOrBT3VLhRdGGO3GblDUppzfL8wsEdoqXjzrJuxSdrGnkFu8S9QjkPn9dCtScvWEcluHqMw=="
+#define MOCK_ECDSA_PUB_KEY "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBK4PZL/iBp7/hQhRvsBh6Mg/UsLXGIRFWTekFq1nJXm3CkBkysa3Jxi9MUkNo0hIDHly2bOMKCKNSpN2U6iGCjw="
```
Comment (allisonkarlitskaya) [inline src/ssh/test-sshbridge.c:702]: The NSA is gonna pwn our CI.

**Review by allisonkarlitskaya — APPROVED** [review]

**Pull request merged**